lib/link.js

```diff
@@ -151,17 +160,15 @@ Link.prototype._detached = function(frame) {
   }
 
   if (this.remoteHandle !== undefined) {
-    this.session._linksByRemoteHandle[this.remoteHandle] = undefined;
+    delete this.session._linksByRemoteHandle[this.remoteHandle];
     this.remoteHandle = undefined;
   }
 
-  this.session._linksByName[this.policy.options.name] = undefined;
-  this.session._allocatedHandles[this.policy.options.handle] = undefined;
-  this.attached = false;
-  this.emit(Link.Detached, { closed: frame.closed, error: frame.error });
+  delete this.session._linksByName[this.policy.options.name];
+  delete this.session._allocatedHandles[this.policy.options.handle];
 
-  // @todo: Session should listen for Link.Detached
-  this.session._linkDetached(this);
+  this.emit(Link.Detached, { closed: frame.closed, error: frame.error });
```

Comment:
It seems like session could potentially listen for Detached, _or_ pass in some callback in createLink, to do this `delete this.session.XXX` work. Thoughts?